From d4a6158e4d4e28d63f2a0f26cf2f357bce1aedaf Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:06:49 -0400
Subject: [powerpc] boot: Properly handle the base "of" boot wrapper

Message-id: <1410545655-205645-181-git-send-email-dzickus@redhat.com>
Patchwork-id: 94180
O-Subject: [RHEL7 PATCH 180/626] powerpc/boot: Properly handle the base "of" boot wrapper
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit 44790a0b93d8481a8dc5bf6aa600941627b56d56
Author: Benjamin Herrenschmidt <benh@kernel.crashing.org>
Date:   Tue Nov 5 10:09:11 2013 +1100

    powerpc/boot: Properly handle the base "of" boot wrapper

    The wrapper script needs an explicit rule for the "of" boot
    wrapper (generic wrapper, similar to pseries). Before
    0c9fa29149d3726e14262aeb0c8461a948cc9d56 it was hanlded
    implicitly by the statement:

    platformo=$object/"$platform".o

    But now that epapr.o needs to be added, that doesn't work
    and an explicit rule must be added.

    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>

Backport notes:  This looks like the revised upsteam fix for RHEL commit
a270a5be5556ec2d707a431c9e71e8eac8d376c8
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/boot/wrapper b/arch/powerpc/boot/wrapper
index c7e74a7..2e1af74 100755
--- a/arch/powerpc/boot/wrapper
+++ b/arch/powerpc/boot/wrapper
@@ -149,6 +149,7 @@ make_space=y
 case "$platform" in
 of)
     platformo="$object/of.o $object/epapr.o"
+    make_space=n
     ;;
 pseries)
     platformo="$object/of.o $object/epapr.o"
-- 
1.7.1